src/com/utils/SudokuGenerator.java

```diff
@@ -0,0 +1,189 @@
+package com.utils;
+
+import com.algorithm.Algorithm;
+import com.algorithm.Backtracking;
+import com.sudoku.Cell;
+import com.sudoku.SudokuBoard;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class that represent the Sudoku Generator class
+ */
+public class SudokuGenerator {
+
+    private static SudokuBoard board = new SudokuBoard();
+
+    public enum Level {
+        EASY,
+        MEDIUM,
+        HARD
+    }
+
+    public static Map<Level, Integer> levels = new HashMap<Level, Integer>();
+
+    static {
+        levels.put(Level.EASY, 3);   // max 3 empty cells per sub grid
```

Comment:
there are 4 levels, you forget the custom one, and the values for each level should be retrieved from the config file